sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-request/_tests_/ecert-request-process-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,198 @@
+        ),
+      );
+      // Student application eligible for e-Cert.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          msfaaNumber,
+          firstDisbursementValues: [
+            createFakeDisbursementValue(
+              DisbursementValueType.CanadaLoan,
+              "CSLP",
+              1122,
+            ),
+          ],
+          institutionLocation: createFakeInstitutionLocation(undefined, {
+            initialValue: { hasIntegration: true },
+          }),
+        },
+        {
+          offeringIntensity: OfferingIntensity.fullTime,
+          applicationStatus: ApplicationStatus.Enrolment,
+        },
+      );
+      application.studentNumber = "1234567789";
+      await db.application.save(application);
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processECERequest(mockedJob.job);
+
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      const fileDate = dayjs().format("YYYYMMDD");
+      expect(result).toEqual(
+        expect.arrayContaining([
+          {
+            summary: [
+              `The uploaded file: ${uploadedFile.remoteFilePath}`,
+              "The number of records: 1",
+            ],
+          },
+        ]),
+      );
+      // Assert file output.
+      const [header, fileDetail, footer] = uploadedFile.fileLines;
+      // Expect the header contains REQUEST and file date.
+      expect(header).toContain(`REQUEST${fileDate}`);
+      const disbursementValue = await db.disbursementValue.findOne({
+        select: {
+          id: true,
+          valueCode: true,
+          valueAmount: true,
+        },
+        where: {
+          disbursementSchedule: {
+            id: In(
+              application.currentAssessment.disbursementSchedules.map(
+                (disbursementSchedule) => disbursementSchedule.id,
+              ),
+            ),
+          },
+        },
+      });
+      // Expect the file detail contains correct data for institution code
+      // disbursement id, disbursement value code, disbursement value amount,
+      // last and first name of the student, application number and student number.
+      expect(fileDetail).toContain(
+        application.currentAssessment.offering.institutionLocation
+          .institutionCode,
+      );
+      expect(fileDetail).toContain(disbursementValue.id.toString());
+      expect(fileDetail).toContain(disbursementValue.valueCode.toString());
+      expect(fileDetail).toContain(disbursementValue.valueAmount.toString());
+      expect(fileDetail).toContain(
+        application.student.user.lastName.slice(0, 25),
+      );
+      expect(fileDetail).toContain(
+        application.student.user.firstName.slice(0, 15),
+      );
+      expect(fileDetail).toContain(application.applicationNumber);
+      expect(fileDetail).toContain(application.studentNumber);
```

Comment:
I would recommend the file detail to assert exact match instead of contain. 

I can be achieved by creating a parser. 

```ts
function buildECEFileDetail(
      application: Application,
      disbursementValue: DisbursementValue,
      disbursementDate: string,
    ): string {
      const DATE_FORMAT = "YYYYMMDD";
      const offering = application.currentAssessment.offering;
      const studyStartDate = formatDate(offering.studyStartDate, DATE_FORMAT);
      const studyEndDate = formatDate(offering.studyEndDate, DATE_FORMAT);
      const disbursementDateFormatted = formatDate(
        disbursementDate,
        DATE_FORMAT,
      );
      const institutionLocation = offering.institutionLocation;
      const student = application.student;
      const studentFirstName = student.user.firstName?.substring(0, 15) ?? "";
      const studentLastName = student.user.lastName.substring(0, 25);
      const studentBirthDate = formatDate(student.birthDate, DATE_FORMAT);
      const institutionStudentNumber = application.studentNumber;
      return `${RecordTypeCodes.ECEDetail}${
        institutionLocation.institutionCode
      }${disbursementValue.id.toString().padStart(10, "0")}${
        disbursementValue.valueCode
      }${disbursementValue.valueAmount.toString().padStart(9, "0")}${
        student.sinValidation.sin
      }${studentLastName.padEnd(25, " ")}${studentFirstName.padEnd(
        15,
        " ",
      )}${studentBirthDate}${
        application.applicationNumber
      }${institutionStudentNumber.padEnd(
        12,
        " ",
      )}100${studyStartDate}${studyEndDate}${disbursementDateFormatted}`;
    }
```

and calling the parser like this. 

```ts
const [disbursement] =
        application.currentAssessment.disbursementSchedules;
      const [disbursementValue] = disbursement.disbursementValues;
      const expectedDetailRecord = buildECEFileDetail(
        application,
        disbursementValue,
        disbursement.disbursementDate,
      );
      expect(fileDetail).toBe(expectedDetailRecord);
```

Reply:
I also tested the result before sharing the above suggestion. 

![image](https://github.com/user-attachments/assets/dc55a949-3ca8-4a7d-b372-8278189a47bc)